pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8scsiv1 "k8s.io/api/storage/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const opConfigCRName = "ceph-csi-operator-config"
+
+func (r *ReconcileCSI) CreateOrUpdateOpConfig(cluster cephv1.CephCluster, clusterInfo *cephclient.ClusterInfo) error {
+	logger.Info("Creating ceph-CSI operator config CR")
+
+	opConfig := &csiopv1a1.OperatorConfig{}
+	opConfig.Name = opConfigCRName
+	opConfig.Namespace = r.opConfig.OperatorNamespace
+	spec := *r.generateCSIOpConfigSpec(cluster, opConfig)
+	opConfig.Spec = spec
+
+	err := clusterInfo.OwnerInfo.SetOwnerReference(opConfig)
```

Comment:
In the next release, we expect that rook will not own updating the operatorConfig CR. It will be fully owned by the user. If we add the ownerref now, we will have to remove it in the next release. So I would vote not to add the ownerref now so we don't have to worry about removing it later. It seems like extra work that doesn't gain us anything in the short term.

Reply:
@Madhu-1 ^^

Reply:
@nb-ohad Your view is that we need the ownerref temporarily. Remind me what benefit you see that would justify the work to remove it later?

Reply:
There are two options here:
1. Add an owner ref and an `Owns` watch to make sure rook gets notified if anyone touches the CR 
2. Add a `Watches` watch with a filter based on name and queue a proper event on the reconciler, this will achieve the same result without an owner ref, and in the next version you will just need to remove the code (and not to touch any resources) 

The major con for option 2 is that you are forfeiting automatic cleanup of these resources when rook needs to clean up and you will need to add manual cleaning code for these resources.

Reply:
@travisn As stated in my reply above, it does not have to be an owner ref. There is always option number 2, that way you don't need to worry about removing the ref in the next upgrade. 

The main point here is that as long as we still treat the CR as an implementing detail and not as a customer-facing CR, we should reconcile out any changes. That means we need to watch for changes.

Reply:
Rook will reconcile the CR again immediately if the `rook-ceph-operator-config` configmap is edited, and also when the rook operator is restarted at some point later. I expect that covers user scenarios since if they want to update the settings, they should be updating the configmap. Otherwise, the user is updating the CR in an unsupported way. What is the urgency to reconcile it immediately? I'm not against adding the watch, it just seems like no need to worry about it.

Reply:
There are a couple of concerns:
1. Undefined behavior over time, if Rook does reconcile, but not immediately (because it does not listen to changes), there will be no way for support or anyone debugging to reason about the state of the reconcile. The question will always be, do we have a bug or is it just that no other event happened that can trigger a reconcile?
2. This goes against the design pattern of controllers: it is part of the design pattern of controllers to react to changes in the desired state and the actual state to try to bring the actual state as close as possible to the desired state. If a change in the actual state does not trigger a reconciliation, and the move toward the desired state is based upon an unrelated/decoupled event then we are breaking the padigram